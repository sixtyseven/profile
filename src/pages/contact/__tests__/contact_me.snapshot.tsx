import React from "react"
import renderer from "react-test-renderer"
import { useStaticQuery } from "gatsby"
import ContactMe from "src/pages/contact/contact_me"

// @ts-ignore
import MockedLayout from "src/components/layout/Layout"
// @ts-ignore
import MockedSEO from "src/components/common/seo"

jest.mock("src/components/layout/Layout", () => {
  return function DummyLayout(props: any) {
    return <div data-testid="layout">{props.children}</div>
  }
})

jest.mock("src/components/common/seo", () => {
  return function DummySEO(props: any) {
    return <div data-testid="seo"></div>
  }
})

beforeEach(() => {
  //@ts-ignore
  useStaticQuery.mockReturnValueOnce({
    site: {
      siteMetadata: {
        linkedInUrl: "linkedInUrl",
      },
    },
  })
})

describe("ContactMe", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<ContactMe />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
