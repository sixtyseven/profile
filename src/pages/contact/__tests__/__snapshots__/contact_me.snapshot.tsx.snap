// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactMe renders correctly 1`] = `
<div
  data-testid="layout"
>
  <div
    data-testid="seo"
  />
  <div
    className="row"
  >
    <div
      className="col-md-8"
    >
      <h2>
        <Styled.code>
          Reach me via: 
        </Styled.code>
      </h2>
      <h4>
        <Styled.code>
          LinkedIn:
           
          <a
            href="linkedInUrl"
            target="_blank"
          >
            linkedInUrl
          </a>
        </Styled.code>
      </h4>
      <h4>
        <Styled.code>
          or Email: lsq991 at gmail dot com
        </Styled.code>
      </h4>
    </div>
  </div>
</div>
`;
